runtime/src/vm/opcode.rs

```diff
@@ -1,25 +1,38 @@
 use std::collections::HashMap;
-use std::convert::TryInto;
+use std::convert::Infallible;
 use std::future::Future;
 use std::hash::Hasher;
+use std::net::SocketAddr;
 use std::pin::Pin;
 use std::process::Command;
-use std::slice;
 use std::str;
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
 use std::time::Duration;
+use std::thread;
+use futures::future::poll_fn;
+use futures::task::{Context, Poll};
 
 use byteorder::{ByteOrder, LittleEndian};
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Request, Response, Server, StatusCode};
+use hyper::header::{HeaderName, HeaderValue};
 use once_cell::sync::Lazy;
+use rand::RngCore;
+use rand::rngs::OsRng;
 use rayon::prelude::*;
 use regex::Regex;
 use tokio::sync::RwLock;
 use tokio::time::delay_for;
 use twox_hash::XxHash64;
 
-use crate::vm::event::{EventEmit, HandlerFragment};
+use crate::vm::event::{BuiltInEvents, EventEmit, HandlerFragment};
 use crate::vm::instruction::InstructionScheduler;
 use crate::vm::memory::{CLOSURE_ARG_MEM_START, HandlerMemory};
+use crate::vm::run::{EVENT_TX};
+
+static HTTP_RESPONSES: Lazy<Arc<Mutex<HashMap<i64, HandlerMemory>>>> = Lazy::new(|| {
```

Comment:
why is a Mutex necessary, wouldn't a RwLock suffice?

Reply:
https://stackoverflow.com/questions/50704279/when-or-why-should-i-use-a-mutex-over-an-rwlock#50704283

I am pretty sure HandlerMemory needs the extra safeguards. I am willing to be proven wrong here, though. I am also expecting us to eventually see this as a bottleneck and switch to dashmap or something like that.